### PR TITLE
Added rowSelection property

### PR DESCRIPTION
**Pull request by Timopheym**
Documentation: https://github.com/adazzle/react-data-grid/blob/0aff0132ae4163fea44391a2f5bc103605dc99cb/docs/markdowns/Viewport.md#rowselection
Code: https://github.com/adazzle/react-data-grid/blob/2b052c986fc47002d84ec7fceedc412f574de484/src/Grid.js#L22

Support configuration as in this example http://adazzle.github.io/react-data-grid/examples.html#/row-select

### CODE REVIEW

File: react-data-grid/react-data-grid.d.ts [hunk 1]
```diff
@@ -173,6 +173,18 @@ declare namespace AdazzleReactDataGrid {
...
+        /**
+         * Options object for selecting rows
+         */
+        rowSelection?: {
+            showCheckbox?: boolean
+            enableShiftSelect?: boolean
+            onRowsSelected?: (rows: Array<Object>) => void,
+            onRowsDeselected?: (rows: Array<Object>) => void,
+            selectBy?: {
+              indexes?: Array<number>
```
Comment (SupernaviX) [inline react-data-grid/react-data-grid.d.ts]: This should also support "keys" and "isSelectedKey"

**Review by SupernaviX — CHANGES_REQUESTED** [review]

**Comment by Timopheym** [comment]
Thanks for Code Review @SupernaviX! I've added your recommendations, is it ok?

**Comment by SupernaviX** [comment]
Just about, the `selectBy` type is slightly off. `keys` should be `{ rowKey: string, values: Array<any> }`, and `isSelectedKey` should be of type `string`. I'd change it myself and save you the trouble, but github doesn't work that way 😜

**Comment by jramsay** [comment]
@Timopheym:  if you can address the final review comments we can merge this PR - thanks!

**Comment by yuit** [comment]
@Timopheym  Thanks for your contribution. This PR has unaddress comment and cannot be merged in at the time being. For housekeeping purposes we are closing stale PRs. Since @SupernaviX  comments should be relatively easy to fix, I will go ahead and open new PR. However, really appreciate you initial PR

**Pull request closed without merging**